src/Messages/DeviceSyncing/OWSSyncGroupsRequestMessage.m

```diff
@@ -0,0 +1,73 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "OWSSyncGroupsRequestMessage.h"
+#import "NSDate+millisecondTimeStamp.h"
+#import "OWSSignalServiceProtos.pb.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface OWSSyncGroupsRequestMessage ()
+
+@property (nonatomic) NSData *groupId;
+
+@end
+
+#pragma mark -
+
+@implementation OWSSyncGroupsRequestMessage
+
+- (instancetype)initWithThread:(nullable TSThread *)thread groupId:(NSData *)groupId
+{
+    self = [super initWithTimestamp:[NSDate ows_millisecondTimeStamp] inThread:thread];
+    if (!self) {
+        return self;
+    }
+
+    OWSAssert(groupId.length > 0);
+    _groupId = groupId;
+
+    return self;
+}
+
+- (void)saveWithTransaction:(YapDatabaseReadWriteTransaction *)transaction
+{
+    // override superclass with no-op.
+    //
+    // There's no need to save this message, since it's not displayed to the user.
+}
+
+- (BOOL)shouldSyncTranscript
+{
+    return NO;
+}
+
+- (BOOL)isLegacyMessage
+{
+    return NO;
```

Comment:
I think this would be the first non-legacy message that is not a sync message.

In theory all the clients should be able to handle this by now, but I think it warrants some extensive testing across other clients.

Alternatively, I think you could make it a legacy message and use the existing `super` implementation of `buildPlainTextData`.